app/models/mixins/authentication_mixin.rb

```diff
@@ -450,4 +464,8 @@ def change_password_params_valid?(current_password, new_password)
 
     raise MiqException::Error, _("Please, fill the current_password and new_password fields.")
   end
+
+  def authentication_class(_authtype)
+    AuthUseridPassword
```

Comment:
Wonder if we could extract [this block](https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/authentication_mixin.rb#L171-L184) to a method that could be shared by this and update_authentication

Reply:
I actually want to get rid of the old implementation eventually, so I'd prefer not to tangle our new code with it. @Fryguy said yesterday that he'd be okay with revising our API for provider creation after DDF is in.

Reply:
My solution is this: https://github.com/ManageIQ/manageiq-providers-openstack/pull/568/files#diff-6f9b687ce51fd31aa102fa9305ef2501R549

Reply:
Okay cool that works too.  So can we convert the `update_authentications` method to use this new method so we don't have two implementations?  Need to keep in mind that the mixin is mixed-in to more than just ExtManagementSystems.

Reply:
Yeah, that's why I'm a little scared to do that :laughing: